### PR TITLE
feat: #390 Add goalMetrics to goals

### PR DESCRIPTION
**Pull request by Ekaterina1994**

### CODE REVIEW

**Review by tritonJS826 — APPROVED** [review]

**Review by SergioAJS — APPROVED** [review]

**Review by tritonJS826 — APPROVED** [review]

File: mw-webapp/src/dataAccessLogic/DTOToBusinessConverter/goalDTOToGoalConverter.ts [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
  */
-export const goalDTOToGoalConverter = (goalDTO: GoalDTO, goalOwner: UserPreview, goalMetric: GoalMetric): Goal => {
+export const goalDTOToGoalConverter = (goalDTO: GoalDTO, goalOwner: UserPreview): Goal => {
   return new Goal({
     ...goalDTO,
     student: goalOwner,
-    metrics: [goalMetric],
+    metrics: goalDTO.metricsStringified.map((metricStringified) => {
+      const metric = JSON.parse(metricStringified);
+      metric.doneDate ? metric.doneDate = new Date(metric.doneDate) : metric.doneDate = null;
```
Comment (tritonJS826) [inline mw-webapp/src/dataAccessLogic/DTOToBusinessConverter/goalDTOToGoalConverter.ts:14]: !

File: mw-webapp/src/logic/wayPage/WayPage.tsx [hunk 2]
```diff
@@ -173,12 +179,22 @@ export const WayPage = (props: WayPageProps) => {
...
   }
 
+  /**
+   * Update way state
+   */
+  const setGoalPartial = (previousGoal: Partial<Goal>) => {
+    if (goal) {
+      const updatedGoal: Goal = {...goal, ...previousGoal};
+      setGoal(updatedGoal);
+    }
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/WayPage.tsx:195]: what if foal is not exist?

File: mw-webapp/src/logic/wayPage/goalBlock/GoalBlock.tsx [hunk 3]
```diff
@@ -114,11 +133,12 @@ export const GoalBlock = (props: GoalBlockProps) => {
             </div>
           </Tooltip>
         </HorizontalContainer>
-        <WayStatistic
+        {/* // TODO: moved out statistics to WayPage because it's not a goal
+         <WayStatistic
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/goalBlock/GoalBlock.tsx:137]: ???

File: mw-webapp/src/logic/wayPage/reportsTable/reportsColumns/ReportsColumns.tsx [hunk 4]
```diff
@@ -104,13 +105,7 @@ export const Columns = (props: ColumnsProps) => {
    * Update DayReport
    * TODO: create generic partialWithUuid
```
Comment (tritonJS826) [inline mw-webapp/src/logic/wayPage/reportsTable/reportsColumns/ReportsColumns.tsx:106]: what should we do?

File: mw-webapp/src/model/DTOModel/GoalDTO.ts [hunk 5]
```diff
@@ -28,6 +23,12 @@ export const GoalDTOSchema = z.object({
    * Estimation time for complete goal
    */
   estimationTime: z.number(),
+
+  /**
+   * Stringified metrics objects
```
Comment (tritonJS826) [inline mw-webapp/src/model/DTOModel/GoalDTO.ts:28]: put a link to something, now this comment does not clarify anything

File: mw-webapp/src/model/businessModelPreview/GoalPreview.ts [hunk 6]
```diff
@@ -28,12 +23,17 @@ export class GoalPreview {
    */
   public estimationTime: number;
 
+  /**
+   * Stringified metrics objects
```
Comment (tritonJS826) [inline mw-webapp/src/model/businessModelPreview/GoalPreview.ts:27]: not clear

**Review by tritonJS826 — APPROVED** [review]

**Pull request merged**